docs/source/references/apollo-config.md

```diff
@@ -0,0 +1,201 @@
+
+**Required** –– the CLI and VS Code extension rely on knowledge of your schema to show you "intellisense" (eg. autocomplete on fields, metrics annotations, query validation).
+
+There are a few different ways you can link your client to a schema:
+1. Use the Apollo [schema registry](/docs/platform/schema-registry.html)
+1. With a remote endpoint (from a running server)
+1. With a local schema file
+
+#### _Option 1_: Use the Apollo schema registry
+
+To link your client to a schema through the Apollo schema registry, you'll need to have at least one version of your schema uploaded to the [registry](/docs/platform/schema-registry.html).
+
+If you have an Engine service set up and tracking your schema's history, you can point your client to that very simply by putting the id of your service in your Apollo config:
+
+```js line=3
+module.exports = {
+  client: {
+    service: 'my-apollo-service', // the id of your service in Engine (from the URL)
+  },
+};
+```
+
+If you're tracking different versions of your schema in the registry using schema tags, you can include the tag you're interested in linking your client to like so:
+
+```js line=3
+module.exports = {
+  client: {
+    service: 'my-apollo-service@beta', // "beta" is the schema tag we're using
+  },
+};
+```
+If a schema tag is not specified, the default value is `current`.
+
+#### Option 2: Link a schema from a remote endpoint
+
+Remote endpoints can be used to pull down a schema from a running service. This can be configured like so:
+
+```js line=3-11
+module.exports = {
+  client: {
+    service: {
+      name: 'github',
+      url: 'https://api.github.com/graphql',
+      // optional headers
+      headers: {
+        authorization: 'Bearer lkjfalkfjadkfjeopknavadf',
+      },
+      // optional disable SSL validation check
+      skipSSLValidation: true,
+    },
+  },
+};
+```
+
+#### Option 3: Link a schema from a local file
+
+In some cases, teams may have a local generated file of their schema they want to link. This can be either a `.graphql` file with the schma in SDL, or a saved introspection result in `.json`. To link a client with a local file, configure the project like this:
+
+```js line=3-6
+module.exports = {
+  client: {
+    service: {
+      name: 'my-service-name',
+      localSchemaFile: './path/to/schema.graphql',
+    },
+  },
+};
+```
+
+### `includes`, `excludes`
+
+*Optional* –– by default, Apollo tools will look under a `./src` directory to find all operations and SDL to extract.
+
+Client projects often contain client-side schema definitions for local state with Apollo Client. To make sure the Apollo CLI and VS Code extension can find these files and read them correctly, you may need to add some additional definitions in your config file. To configure which folders to include, and which to ignore, adjust your config like so:
+
+```js line=3-4
+module.exports = {
+  client: {
+    includes: ['./imports/**/*.js'],
+    excludes: ['**/__tests__/**/*'],
```

Comment:
We should note the `excludes` default somewhere as well:
`excludes: ["**/node_modules", "**/__tests__"]`